substrate/frame/child-bounties/src/lib.rs

```diff
@@ -181,16 +189,22 @@ pub mod pallet {
 		Canceled { index: BountyIndex, child_index: BountyIndex },
 	}
 
-	/// Number of total child bounties.
+	/// DEPRECATED: Replaced with `ParentTotalChildBounties` storage item keeping dedicated counts
+	/// for each parent bounty. Number of total child bounties.
 	#[pallet::storage]
 	pub type ChildBountyCount<T: Config> = StorageValue<_, BountyIndex, ValueQuery>;
 
-	/// Number of child bounties per parent bounty.
+	/// Number of active child bounties per parent bounty.
 	/// Map of parent bounty index to number of child bounties.
 	#[pallet::storage]
 	pub type ParentChildBounties<T: Config> =
```

Comment:
Are we sure this can be deprecated? This storage item enables `O(1)` lookup in this line
```
			ensure!(
				ParentChildBounties::<T>::get(parent_bounty_id) <=
					T::MaxActiveChildBountyCount::get() as u32,
				Error::<T>::TooManyChildBounties,
			);
```
Without this we'd have to count all the prefix items in `ChildBounties` storage item to get the result of active bounties.